[#1] Chore: 웹팩 및 바벨 설정

sageherb opened the pull request:
## 개요

이슈 번호: #1 

웹팩 및 바벨 설정

## 작업 내용

- [x] 웹팩 및 바벨 디펜던시 설치
- [x] 웹팩 설정

## 기타

html 파일에 이미지 주소(`<img src="./aseets/image.svg" />`)를 사용하면 빌드 시 전부 정상적으로 작동하지만, 개발 서버에선 파비콘이 아예 뜨지 않고, 이미지도 처음엔 잘 뜨다가 재컴파일을 한 후에 이미지 주소가 404 에러가 뜨는 문제가 있습니다.
주소에 `<%= require() %>`을 사용해도 같은 문제가 발생했습니다.
해당 문제 해결을 위해 계속 서치해봤으나 `HttpWebpackPlugin` 버그로 예상됩니다.
[관련 HttpWebpackPlugin 이슈 글](https://github.com/jantimon/html-webpack-plugin/issues/1709)

sageherb committed:
Chore: 프로젝트 생성 및 웹팩 셋업

f-lab-liv commented on .gitignore:
```diff
@@ -0,0 +1,211 @@
+# Created by https://www.toptal.com/developers/gitignore/api/macos,visualstudiocode,node,git
+# Edit at https://www.toptal.com/developers/gitignore?templates=macos,visualstudiocode,node,git
```
git ignore 생성해주는 사이트는 참고용으로 사용을 하고, 실제로 필요한 값만 사용하는 것이 좋을 것 같습니다.

sageherb replied:
`.gitignore` 수정했습니다!

sageherb committed:
Chore: .gitignore 수정

f-lab-liv reviewed (approved):
webpack 설정은 일단 이슈 없어보입니다! 이미지 관련해서 이슈 있다고 하신 부분 어떤 건지 잘 모르겠는데, 다시 한 번 어떤 상황이신지 알려주시면 확인해보겠습니다.

다만 현재 development 환경 기준으로 세팅하신 것 같은데, 추후 production과 세팅을 분리하고, production에는 추가적으로 어떤 설정들이 필요할지 찾아보시면 좋을 것 같습니다.

sageherb committed:
Chore: 웹팩 프로덕션 개발 옵션 분리

- 공통 옵션인 webpack.common.js 생성
- 개발 옵션인 webpack.dev.js 생성
- 프로덕션 옵션인 webpack.prod.js 생성
- npm build, dev 옵션 수정
- html-loader 디펜던시 추가
- webpack-merge 디펜던시 추가

f-lab-liv commented on webpack.prod.js:
```diff
@@ -0,0 +1,6 @@
+const { merge } = require("webpack-merge");
+const common = require("./webpack.common.js");
+
+module.exports = merge(common, {
+  mode: "production",
+});
```
production에 아무런 추가적인 설정이 없는데, 시간 나실 때 production에는 어떤 설정이 들어가면 좋을지 보시면 좋을 것 같습니다.

sageherb replied:
넵! 작업하면서 틈틈이 프로덕션 설정 찾아보겠습니다!

f-lab-liv commented:
@sageherb 
다음 작업에 blocker가 되지 않길 바래서 apporove 드렸습니다. 이 pr은 머지하셔도 괜찮습니다!
https://github.com/f-lab-edu/vanilla-spa-clone/pull/2#discussion_r1452319926 부분만 시간 나실 때 확인해주세요!
